src/libcore/ptr.rs

```diff
@@ -10,9 +10,65 @@
 
 // FIXME: talk about offset, copy_memory, copy_nonoverlapping_memory
 
-//! Raw, unsafe pointers, `*const T`, and `*mut T`.
+//! Manually manage memory through raw pointers.
 //!
 //! *[See also the pointer primitive types](../../std/primitive.pointer.html).*
+//!
+//! # Safety
+//!
+//! Many functions in this module take raw pointers as arguments and dereference
+//! them. For this to be safe, these pointers must be valid. However, because
+//! rust does not yet have a formal memory model, determining whether an
+//! arbitrary pointer is valid for a given operation can be tricky.
+//!
+//! There are two types of operations on memory, reads and writes. A single
+//! pointer can be valid for any combination of these operations. For example, a
+//! pointer is not valid for writes if a `&mut` exists which [refers to the same
+//! memory][aliasing]. The set of operations for which a pointer argument must
+//! be valid is explicitly documented for each function. This is not strictly
+//! necessary for `*const` arguments, as they can only be used for reads and
+//! never for writes.
```

Comment:
I am fine with the content of this, but the way this is phrased It may be just me but reading this paragraph leaves me pretty confused. For example, the "set of operations" can not be larger than 2, ever; and I also never find myself wondering "what is the set of operations that this pointer is valid for". I rather wonder "can I read through this"? Also, I do not think it is possible for a pointer to be valid for writing, but not valid for reading -- so "any combination" is not really possible.

So, I'd frame this entire thing by starting out saying there are *two* notions of validity: Valid for reading, and valid for writing. Every pointer operation specifies which kind of validity it requires. I'd put all of that into the first paragraph, and *not* talk about how Rust (typo: rust) doesn't yet have a fully defined model there. So, the first paragraph sets the stage and introduced the distinction between read and write validity, and does nothing else.

The second paragraph can then lead by saying that unfortunately, Rust doesn't yet have a fully defined memory model and hence the precise set of rules is not known yet. However, there are some things we do know -- e.g., if an `&mut` exists *and is not currently reborrowed* (that restriction is important), then any other aliasing pointer is not valid for reading nor writing. And so on.

It might even be better to move this paragraph down; the sentence right before your enumeration also touches the same topic so that would flow nicely.